devito/operations/interpolators.py

```diff
@@ -388,3 +389,520 @@ def callback():
+            # the right side of the equation
+            summands.extend([Inc(rhs, v, implicit_dims=self.sfunction.dimensions)
+                             for v in eqs])
+
+            # Write/Incr `self`
+            lhs = self.sfunction.subs(self_subs)
+            last = [Inc(lhs, rhs)] if increment else [Eq(lhs, rhs)]
+
+            return temps + summands + last
+
+        return Interpolation(expr, offset, increment, self_subs, self, callback)
+
+    def inject(self, field, expr, offset=0):
+        """
+        Generate equations injecting an arbitrary expression into a field.
+        Parameters
+        ----------
+        field : Function
+            Input field into which the injection is performed.
+        expr : expr-like
+            Injected expression.
+        offset : int, optional
+            Additional offset from the boundary.
+        """
+        def callback():
+            # Derivatives must be evaluated before the introduction of indirect accesses
+            try:
+                _expr = expr.evaluate
+            except AttributeError:
+                # E.g., a generic SymPy expression or a number
+                _expr = expr
+
+            variables = list(retrieve_function_carriers(_expr)) + [field]
+
+            # Need to get origin of the field in case it is staggered
+            field_offset = field.origin
+
+            # List of indirection indices for all adjacent grid points
+            idx_subs, temps = self._interpolation_indices(variables, offset,
+                                                          field_offset=field_offset)
+
+            dim_pos = self.sfunction.grid.dimensions
+            if len(dim_pos) == 3:
+                eqs = [v for v in self._tricubic_equations(_expr,
+                                                           idx_subs,
+                                                           dim_pos=dim_pos)]
+            else:
+                eqs = [v for v in self._bicubic_equations(_expr,
+                                                          idx_subs,
+                                                          dim_pos=dim_pos)]
+
+            size = np.prod(np.shape(eqs))
+            eqs = [e[0] for e in np.reshape(eqs, (size, 1))]
+
+            eqns = [Inc(field.xreplace(vsub), b,
+                        implicit_dims=self.sfunction.dimensions)
+                    for b, vsub in zip(eqs, idx_subs)]
+
+            return temps + eqns
+
+        return Injection(field, expr, offset, self, callback)
+
+
+class SincInterpolator(GenericInterpolator):
+    """
+    Concrete implementation of GenericInterpolator implementing a Sinc interpolation
+    scheme, using a method that is windowed by a Kaiser Function.
+    Parameters
+    ----------
+    sfunction: The SparseFunction that this Interpolator operates on.
+    """
+
+    def __init__(self, sfunction):
+        self.sfunction = sfunction
+        self.r = 4
```

Comment:
This should probably be a class property

Reply:
Is it a problem to let the user choose?

Reply:
> Is it a problem to let the user choose?

The Bessel function was implemented inside the .h file considering `r = 4`. So it would be a problem to allow the user to choose the value of r